### DISCUSSION TITLE
1 validation error detected: Value '' at 'shipmentType' failed to satisfy constraint: Member must satisfy enum value set: [LTL, SP]"

### DISCUSSION THREAD

**Discussion by IncredibleX** [abuzuhri/Amazon-SP-API-CSharp#149] [Q&A] [answered]
Hi,
I got the errormessage although there is a value for ShipmentType.
<img width="500" alt="Bildschirmfoto 2022-03-24 um 11 41 35" src="https://user-images.githubusercontent.com/38632368/159900142-3c9c79a1-2b94-4d67-959c-24b56b97cc87.png">

If I Debug PutTransportDetails.cs, I can see, that the ShipmentType Value is missing after the Json Serialization.
<img width="1077" alt="Bildschirmfoto 2022-03-24 um 11 42 21" src="https://user-images.githubusercontent.com/38632368/159900054-a7c0734e-a78b-4fc3-b993-d2e20d51be11.png">

Code:
`        oPutTransportDetailsRequest.TransportDetails = new FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.TransportDetailInput();
                    

                    FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.PartneredSmallParcelPackageInput oPackageInputItem = new FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.PartneredSmallParcelPackageInput();
                    oPackageInputItem.Weight = new FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.Weight();
                    oPackageInputItem.Weight.Value = 10;
                    oPackageInputItem.Weight.Unit = FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.UnitOfWeight.Kilograms;
                    
                    oPackageInputItem.Dimensions = new FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.Dimensions();
                    oPackageInputItem.Dimensions.Width = 50;
                    oPackageInputItem.Dimensions.Height = 50;
                    oPackageInputItem.Dimensions.Length = 50;
                    oPackageInputItem.Dimensions.Unit = FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.UnitOfMeasurement.Centimeters;

                    oPutTransportDetailsRequest.TransportDetails.PartneredSmallParcelData = new FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.PartneredSmallParcelDataInput();
                    oPutTransportDetailsRequest.TransportDetails.PartneredSmallParcelData.PackageList = new FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.PartneredSmallParcelPackageInputList();
                    oPutTransportDetailsRequest.TransportDetails.PartneredSmallParcelData.PackageList.Add(oPackageInputItem);
                    oPutTransportDetailsRequest.ShipmentType = FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.ShipmentType.SP;
                    FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.CommonTransportResult oResult = amazonConnection.FulFillmentInbound.PutTransportDetails(sShipmentID, oPutTransportDetailsRequest);
`

**Comment by IncredibleX** [comment, marked as answer]
In Amazon-SP-API-CSharp-main\Source\FikaAmazonAPI\AmazonSpApiSDK\Models\FulfillmentInbound\PutTransportDetailsRequest.cs in FikaAmazonApi

as Workaround I changed ShipmentType from enum to string, after that it worked.
'        /// <summary>
        /// Gets or Sets ShipmentType
        /// </summary>
        [DataMember(Name = "ShipmentType", EmitDefaultValue = false)]
        public string ShipmentType { get; set; }
        /// <summary>
        /// Initializes a new instance of the <see cref="PutTransportDetailsRequest" /> class.
        /// </summary>'

**Reply by ouzhiman** [reply]
Thanks, it's ok to change it to a string

**Comment by abuzuhri** [comment]
I fix this please see this commit 

https://github.com/abuzuhri/Amazon-SP-API-CSharp/commit/ba5b9c0d3efea1f59c8c357824b0dfd468f0446b

**Comment by IncredibleX** [comment]
Your solution looks better than my workaround…but it doesn’t work:


Tobias Stephan
eCommerce & Softwaredevelopment
+49 171 3833568
***@***.***
Shops:
www.leckerstecker.de
www.herr-der-dinge.de

Softwareblog:
www.exoda.de
Kookamp 40, 46354 Südlohn
Apps:
Leckerstecker Kabelshop
3D Print for Dremel 3D45 3D40
Listory Lister Shopping App
MCR-Tools

...noch mehr

Am 24. März 2022, 16:18 +0100 schrieb Tareq Abuzuhri ***@***.***>:

**Comment by abuzuhri** [comment]
Please open new issue and add the code you use to it and error message.

I will look to all issues this weekend
